src/librustdoc/html/highlight.rs

```diff
@@ -154,10 +154,57 @@ fn get_real_ident_class(text: &str, edition: Edition) -> Class {
     }
 }
 
+/// This iterator comes from the same idea than "Peekable" except that it allows to "peek" more than
+/// just the next item by using `peek_next`. The `peek` method always returns the next item after
+/// the current one whereas `peek_next` will return the next item after the last one peeked.
+///
+/// You can use both `peek` and `peek_next` at the same time without problem.
+struct PeekIter<'a> {
```

Comment:
There's no need to rewrite this. Just use `Peekable<Peekable>`.

Reply:
Doesn't it only allow to check the two next items?

Reply:
Oh I see, this allows peeking arbitrarily many times. I guess that's ok, it seems like it could be expensive to store all those tokens in memory at once though.

Reply:
Unless you have a lot of whitespaces, the size of stored items should never go higher than 2. But just in case... :)

Reply:
If the memory consumption becomes concerning in any case, then you could do the equivalent of "clone current iterator, advance N times, return that value and drop the cloned iterator" to get the same functionality at the cost of recomputing the value (advancing the iterator) every time... You could even add logic to either delegate to `Peekable<Peekable<T>>` or do the above depending on the N.